src/Build/Instance/ProjectInstance.cs

```diff
@@ -2940,7 +2940,7 @@ private static ProjectPropertyInstance InstantiateProjectPropertyInstance(Projec
             _itemDefinitions = new RetrievableEntryHashSet<ProjectItemDefinitionInstance>(MSBuildNameIgnoreCaseComparer.Default);
             _hostServices = buildParameters.HostServices;
             this.ProjectRootElementCache = buildParameters.ProjectRootElementCache;
-            _loggingContext = new AnalyzerLoggingContext(loggingService, buildEventContext);
+            _loggingContext = new AlwaysValidLoggingContext(loggingService, buildEventContext);
```

Comment:
why does Expander need `AnalyzerLoggingContext`?

Reply:
It needs any loggingContext. This one was just only one creatable with availalbel info (we cannot create node/project/target/...-LoggingContext here)

As to why - it was historically accepting LoggingContext as optional parameter in it's numerous methods - so the control flow and compexity of the code was unnecesarily complicated. In order to be able to transfer info from evauation (and expansion) from Evaluator (and Expander) to the BuildCheck (e.g. acquisition request expressed via property function, or the info abourt property reads/writes) - we started to need LoggingContext deterministicaly.